### PR TITLE
added conditions to enhance  pip command parsing to support git urls

### PR DESCRIPTION
**Pull request by jubril-adeyi**
### Description

This pull request imporoves the InstallParser class to improve the handling of pip dependencies . It specifically adds support for Git URLs and wheel files, addressing a limitation in handling such  edge cases in dependency installation 

### Changes to Be Made
Updated the command parsing logic to:

-  Accept two or three arguments for pip install commands.
-  Allow Git URLs without version specification.
-  Allow wheel files without additional arguments.

I implemented the assert to ensure the number of argument and maintained theerror handling.
Related to #21

### CODE REVIEW

**Comment by jubril-adeyi** [comment]
@DhanshreeA I made some changes to the /src/parser.py file  to handle pip dependency installations for git URLs and wheel files/urls. 

I'd like to know if I am on the right path with this implementation
Also Should I also add handling for cases where it is a regular pip install command and doesn’t specify a version?

**Comment by DhanshreeA** [comment]
@jubril-adeyi Technically, pip dependencies can be specified in several different way. From the official [documentation](https://pip.pypa.io/en/stable/reference/requirement-specifiers/#examples), these are some of the ways that "named" pip dependencies are specified,

> ```
> SomeProject
> SomeProject == 1.3
> SomeProject >= 1.2, < 2.0
> SomeProject[foo, bar]
> SomeProject ~= 1.4.2
> SomeProject == 5.4 ; python_version < '3.8'
> SomeProject ; sys_platform == 'win32'
> requests [security] >= 2.8.1, == 2.8.* ; python_version < "2.7"
> ```

And this is how URL based pip dependencies are specified:

> ```
> pip @ https://github.com/pypa/pip/archive/22.0.2.zip
> requests [security] @ https://github.com/psf/requests/archive/refs/heads/main.zip ; python_version >= "3.11"
> ```

In all of these cases except "`SomeProject`", and "`SomeProject[foo, bar]`" there is a clear version to the dependency being specified. 
What we really want is to always ensure reproducibility, to enforce our model contributors to specify dependencies as either:

1. SomeProject == 1.3; or
2. pip @ https://github.com/pypa/pip/archive/22.0.2.zip
3. pip install git+https://github.com/bp-kelley/descriptastorus.git@d552f934757378a61dd1799cdb589a864032cd1b
4. pip install torchvision==0.17.1  --index-url https://download.pytorch.org/whl/cpu

This change will go both in the YAMLInstallParser and the DockerfileInstallParser. If you notice, the DockerfileInstallParser technically just converts a Dockerfile into the same format as if it were a YAMLInstall file. So ideally what we want is the following:

1. In the case of `install.yml` file, users should be able to specify dependencies in one of the following ways:

- `['pip', 'SomeProject', '1.3']`
- `['pip', 'git', 'https://github.com/bp-kelley/descriptastorus.git', 'd552f934757378a61dd1799cdb589a864032cd1b']`
- `['pip', 'torchvision', '0.17.1', '--index-url', 'https://download.pytorch.org/whl/cpu']`

To summarize, either we need a version, or link to a VCS repo (git is a good starting point) and a commit sha so we don't end up installing always from the main branch, or a version + additional commands like an index-url.

Hope this helps!

**Comment by DhanshreeA** [comment]
Closing because this is not a satisfactory solution and does not account for all the cases.
